packages/patternfly-4/react-core/src/layouts/Page/Page.js

```diff
@@ -0,0 +1,113 @@
+
+export const PageLayouts = {
+  vertical: 'vertical',
+  horizontal: 'horizontal'
+};
+
+const propTypes = {
+  /** Content rendered inside the Page layout */
+  children: PropTypes.node,
+  /** Additional classes added to the Page layout */
+  className: PropTypes.string,
+  /** Page layout type */
+  layout: PropTypes.oneOf(Object.values(PageLayouts)),
+  /** Component to render the background image */
+  background: PropTypes.node,
+  /** Component to render the logo/brand */
+  logo: PropTypes.node,
+  /** Component to render the navigation */
+  nav: PropTypes.node,
+  /** Vertical layout only: True if the nav is expanded  */
+  isNavOpen: PropTypes.bool,
+  /** Vertical page layout only: Callback function to handle the navigation toggle button */
+  onNavToggle: PropTypes.func,
+  /** Component to render the toolbar */
+  toolbar: PropTypes.node,
+  /** Component to render the avatar */
+  avatar: PropTypes.node
+};
+
+const defaultProps = {
+  children: null,
+  className: '',
+  layout: 'vertical',
+  background: null,
+  logo: 'Logo',
+  nav: 'Nav',
+  onNavToggle: () => undefined,
+  isNavOpen: true,
+  toolbar: 'Toolbar',
+  avatar: null
+};
+
+const Page = ({
+  className,
+  children,
+  layout,
+  background,
+  logo,
+  nav,
+  isNavOpen,
+  onNavToggle,
+  toolbar,
+  avatar,
+  ...props
+}) => (
+    <React.Fragment>
+      {background}
+      <div {...props} className={css(styles.page, className)}>
+        {/* Added style={{ overflowX: 'unset' }} as a workaround to make dropdowns work */}
+        <header role="banner" className={css(styles.pageHeader)} style={{ overflowX: 'unset' }}>
+          <div className={css(styles.pageHeaderBrand)}>
+            {layout === PageLayouts.vertical && (
+              <div className={css(styles.pageHeaderBrandToggle)}>
+                <Button
+                  id="nav-toggle"
+                  onClick={onNavToggle}
+                  aria-label="Toggle primary navigation"
+                  variant={ButtonVariant.plain}
+                >
+                  <BarsIcon />
+                </Button>
+              </div>
+            )}
+            <a className={css(styles.pageHeaderBrandLink)}>{logo}</a>
+          </div>
+          {/* Hide for now until we have the context selector component */}
+          {/* <div className={css(styles.pageHeaderSelector)}>
+            pf-c-context-selector
+          </div> */}
+          {layout === PageLayouts.horizontal && <div className={css(styles.pageHeaderNav)}>{nav}</div>}
+          <div className={css(styles.pageHeaderTools)}>
+            {toolbar}
+            {avatar}
+          </div>
+        </header>
+        {layout === PageLayouts.vertical && (
+          <aside
+            className={css(
+              styles.pageSidebar,
+              isNavOpen && styles.modifiers.expanded,
+              !isNavOpen && styles.modifiers.collapsed
+            )}
+          >
+            {nav}
```

Comment:
Nesting expandable nav inside the `<aside>` has an issue with the chevron icons, tracked here:
https://github.com/patternfly/patternfly-next/issues/779